### PR TITLE
Added test for similarity score

### PR DESCRIPTION
**Pull request by kathirsvn**
**What this PR does**:

- Fixed `projection` in `findOne`
- Added test for similarity score in `find` and `findOne` APIs

**Checklist**
- [X] Changes manually tested
- [X] Automated Tests added/updated
- [X] Documentation added/updated
- [X] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: tests/driver/api.test.ts [hunk 1]
```diff
@@ -807,6 +807,19 @@ describe('Mongoose Model API level tests', async () => {
             assert.deepStrictEqual($vector, [1, 101]);
         });
 
+        it('supports sort() and similarity score with $meta with find()', async function() {
+            const res = await Vector.find({}, { name: 1, $similarity : 1}).sort({ $vector: { $meta: [1, 99] } });
```
Comment (vkarpov15) [inline tests/driver/api.test.ts:811]: What do I need to do to get these tests to pass? I'm running locally with latest JSON API, but this test fails with the following error for me.

```
Error: Command "find" failed with the following errors: [{"message":"Unsupported projection parameter: path cannot start with '$' (no root-level operators)","errorCode":"UNSUPPORTED_PROJECTION_PARAM","exceptionClass":"JsonApiException"}]
```

File: tests/driver/api.test.ts [hunk 2]
```diff
@@ -807,6 +807,19 @@ describe('Mongoose Model API level tests', async () => {
...
+            assert.deepStrictEqual(res.map(doc => doc.name), ['Test vector 1', 'Test vector 2']);
+            assert.deepStrictEqual(res.map(doc => doc.get('$similarity')), [1, 0.51004946]);
```
Comment (vkarpov15) [inline tests/driver/api.test.ts:813]: This `doc.get()` is a bit clunky, but you're right that's the only way you can access a property that's returned from the database but not in your schema.

One potential alternative is to use a virtual for `$similarity` as follows:

```javascript
  schema.virtual('$similarity').
    get(function() { return this._doc.$similarity; });
```

That will allow you to do `doc.$similarity` without needing to use `get()`.

File: tests/driver/api.test.ts (hunk 1 above)
Comment (kathirsvn) [reply tests/driver/api.test.ts:811]: I'm running with below versions in `api-compatibility.versions` and starting the server using `bin/start_json_api.sh` and it's supported.
```
stargate_version=v2.1.0-ALPHA-6
json_api_version=v1.0.0-ALPHA-12
```

File: tests/driver/api.test.ts (hunk 2 above)
Comment (kathirsvn) [reply tests/driver/api.test.ts:813]: I was not aware, we could use `virtual` for this. Good to know, thanks!

**Review by vkarpov15 — APPROVED** [review]
LGTM :+1:

**Pull request merged**